src/libhirte/common/cfg.c

```diff
@@ -338,3 +339,95 @@ const char *cfg_dump(struct config *config) {
         }
         return cfg_info;
 }
+
+int cfg_agent_def_conf(struct config *config) {
+        int result = 0;
+
+        result = cfg_set_default_section(config, CFG_SECT_AGENT);
+        if (result != 0) {
+                fprintf(stderr,
+                        "Error setting default section for agent '%s', error code '%s'.\n",
+                        CFG_SECT_AGENT,
+                        strerror(-result));
+                return false;
```

Comment:
```suggestion
                return result;
```
The caller is already doing a `fprintf`.